### PR TITLE
[SPARK-49451][FOLLOW-UP] Add support for duplicate keys in from_json(_, 'variant')

### PR DESCRIPTION
**Pull request by harshmotw-db**
### What changes were proposed in this pull request?

This PR adds support for duplicate key support in the `from_json(_, 'variant')` query pattern. Duplicate key support [has been introduced](https://github.com/apache/spark/pull/47920) in `parse_json`, json scans and the `from_json` expressions with nested schemas but this code path was not updated.

### Why are the changes needed?

This change makes the behavior of `from_json(_, 'variant')` consistent with every other variant construction expression.

### Does this PR introduce _any_ user-facing change?

It potentially allows users to use the `from_json(<input>, 'variant')` expression on json inputs with duplicate keys depending on a config.

### How was this patch tested?

Unit tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by harshmotw-db** [comment]
@chenhao-db @cloud-fan This PR follows up on [the previous PR](https://github.com/apache/spark/pull/47920) that added support for duplicate keys where one corner case was left out. Can you review this PR? Thanks!

**Review by MaxGekk — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala [hunk 1]
```diff
@@ -717,9 +717,12 @@ case class JsonToStructs(
   override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression =
     copy(timeZoneId = Option(timeZoneId))
 
+  private val variantAllowDuplicateKeys = SQLConf.get.getConf(SQLConf.VARIANT_ALLOW_DUPLICATE_KEYS)
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala:720]: Actually, this is not proper approach of getting the SQL config. The expression can change its behaviour during analys, see https://github.com/apache/spark/pull/27658

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala (hunk 1 above)
Comment (harshmotw-db) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala]: Thanks for pointing this out! It has been fixed now.

**Review by cloud-fan — APPROVED** [review]

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**